woocommerce/woocommerce-admin
# Replace marketing extension - Google Listings and Ads

Fixes 5-gh-woocommerce/google-listings-and-ads

This PR replaces the previous Google Ads extension with the Google Listings and Ads extension.

### Testing instructions
View the installed marketing extensions section on the page Marketing > Overview 
The extension should appear in the following states:

1. Extension not installed (should not be included in the list, if the list is empty this section won't show at all)
![image](https://user-images.githubusercontent.com/11388669/117135419-d2529e80-ad9e-11eb-9752-081f00beb11c.png)

2. Extension installed but not activated (click activate to confirm it works)
![image](https://user-images.githubusercontent.com/11388669/117135541-fb732f00-ad9e-11eb-95f9-d3739fc715da.png)

3. Extension installed and activated but not setup yet (click "Finish setup" to confirm we get redirected to the get started page)
![image](https://user-images.githubusercontent.com/11388669/117135812-602e8980-ad9f-11eb-83f8-29889cdb7202.png)

4. Extension installed, activated and setup (click settings to confirm we get redirected to the settings page)
![image](https://user-images.githubusercontent.com/11388669/117135999-ab489c80-ad9f-11eb-874b-5f6f5ec4ce5d.png)

Note: The documentation link is not active yet (still in draft)

#### Test onboarding
1. Make sure the extension is installed but not activated
2. [Reset](https://github.com/woocommerce/woocommerce-admin/wiki/Common-Issues#reset-the-onboarding-wizard) the onboarding wizard, and go to `wp-admin/admin.php?page=wc-admin&path=%2Fsetup-wizard`
3. Go through all the onboarding steps till we reach `Included business features`
4. Ensure the plugin shows up correctly with the right naming and select it as the only extension to add
![image](https://user-images.githubusercontent.com/11388669/117138107-7be75f00-ada2-11eb-9077-c839cee2155e.png)
5. Continue and confirm the `plugins were successfully activated` notice shows up
6. Finish onboarding and confirm that the extension was actually activated

Note: Ideally we should test the same onboarding flow, without the extension installed, however at the moment the extension with slug `google-listings-and-ads` is not live on WordPress.org yet (review process is still pending).

#### Test old note removal
If you do not currently have the old note on your WooCommerce Admin home screen, you can add a test note with the correct name as follows:
1. install the WooCommerce Admin Test Helper plugin [here](https://github.com/woocommerce/woocommerce-admin-test-helper)
2. Go to the Admin notes tab
3. Add an admin note with the name `wc-admin-google-ads-and-marketing`
4. Go to the WCA home screen and verify that your test note is present

The note is removed on a new version install, so either install an old version of WCA and upgrade to the current one, or trigger the install process manually:
1. install the WCA test helper
2. go to the Tools tab
3. click the `Trigger WCA install` button


### Changelog note
- Update: Replace marketing extension - Google Listings and Ads.
